docs/build.md

```diff
@@ -0,0 +1,120 @@
+# Builds
+
+A Build is a resource object to schedule and run a single [Cloud Native Buildpacks](http://buildpacks.io) build.
+
+Builds provide more granular control than the [Image resource](image.md). However, you will need to manually specify the source, and the cache volume. Additionally, you will need to create a new build for every build execution as builds will not rebuild on source code and buildpack updates. 
+
+### Configuration
+
+```yaml
+apiVersion: build.pivotal.io/v1alpha1
+kind: Build
+metadata:
+  name: sample-build
+spec:
+  tags:
+  - sample/image
+  serviceAccount: service-account
+  builder:
+    image: cloudfoundry/cnb:bionic
+    imagePullSecrets: 
+    - name: builder-secret  
+  cacheName: persisent-volume-claim-name
+  source:
+    git:
+      url: https://github.com/buildpack/sample-java-app.git
+      revision: master
+  env:
+  - name: "JAVA_BP_ENV"
+    value: "value"
+  resources:
+    limits:
+      cpu: "0.25"
+      memory: "128M"
+    requests:
+      cpu: "0.5"
+      memory: "256M"
+```
+
+- `tags`: A list of docker tags to build. At least one tag is required.
+- `serviceAccount`: The Service Account name that will be used for credential lookup. Check out the [secrets documentation](secrets.md) for more information. 
+- `builder.image`: This is the tag to the [Cloud Native Buildpacks builder image](https://buildpacks.io/docs/using-pack/working-with-builders/) to use in the build. Unlike on the Image resource, this is an image not a reference to a Builder resource.    
+- `builder.imagePullSecrets`: An optional list of pull secrets if the builder is in a private registry. [To create this secret please reference this link](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials)
+- `source`: The source location that wil be the input to the build. See the [Source Configuration](#source-config) section below.
+- `cacheName`: Optional name of a persistent volume claim to used for a build cache across builds.
+- `env`: Optional list of build time environment variables.
+- `resources`: Optional configurable resource limits on `CPU` and `memory`.
+
+> Note: All fields on a build are immutable. Instead of updating a build, create a new one.
```

Comment:
This seems quite important.  Maybe move above this section in which you define the purpose of each field.

Reply:
Mentioned immutability at the beginning.